File: masiwasi/app/src/main/java/adapters/MascotaAdapter.java
package adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

//import com.bumptech.glide.Glide;
import com.example.masiwasi.R;

import java.util.List;

import models.Mascota;

public class MascotaAdapter extends RecyclerView.Adapter<MascotaAdapter.MascotaViewHolder> {

    private List<Mascota> mascotaList;
    private OnMascotaClickListener listener;
    private Context context;
    private boolean modoEdicion;

    public interface OnMascotaClickListener {
        void onVerDetalles(Mascota mascota);
    }

    public MascotaAdapter(Context context, List<Mascota> mascotaList, boolean modoEdicion, OnMascotaClickListener listener) {
        this.context = context;
        this.mascotaList = mascotaList;
        this.listener = listener;
        this.modoEdicion = modoEdicion;
    }

    public void setModoEdicion(boolean modoEdicion) {
        this.modoEdicion = modoEdicion;
        notifyDataSetChanged(); // Actualiza el botón de cada item
    }

    @NonNull
    @Override
    public MascotaViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_mascota, parent, false);
        return new MascotaViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull MascotaViewHolder holder, int position) {
        Mascota mascota = mascotaList.get(position);

        holder.txtNombre.setText(mascota.getNombre());
        holder.txtEdad.setText(mascota.getEdad());
        holder.txtSexo.setText(mascota.getSexo());
        holder.txtDescripcion.setText(mascota.getDescripcion());

//        Glide.with(context)
//                .load(mascota.getImageUrl())
//                .placeholder(R.drawable.ic_launcher_foreground)
//                .into(holder.imgMascota);

        holder.btnDetalles.setText(modoEdicion ? "Editar" : "Ver/Detalles");

        holder.btnDetalles.setOnClickListener(v -> {
            if (listener != null) listener.onVerDetalles(mascota);
        });
    }

    @Override
    public int getItemCount() {
        return mascotaList.size();
    }

    public static class MascotaViewHolder extends RecyclerView.ViewHolder {

        ImageView imgMascota;
        TextView txtNombre, txtEdad, txtSexo, txtDescripcion;
        Button btnDetalles;

        public MascotaViewHolder(@NonNull View itemView) {
            super(itemView);
            imgMascota = itemView.findViewById(R.id.imgMascota);
            txtNombre = itemView.findViewById(R.id.txtNombreMascota);
            txtEdad = itemView.findViewById(R.id.txtEdadMascota);
            txtSexo = itemView.findViewById(R.id.txtSexoMascota);
            txtDescripcion = itemView.findViewById(R.id.txtDescripcionMascota);
            btnDetalles = itemView.findViewById(R.id.btnVerDetalles);
        }
    }
}

File: movil2/mozita/app/src/main/java/adapters/MascotaAdapter.kt
package adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import pe.edu.idat.mozitaapp.R
import models.Mascota

class MascotaAdapter(
    private val context: Context,
    private var mascotaList: MutableList<Mascota>,
    private var modoEdicion: Boolean,
    private val listener: OnMascotaClickListener?
) : RecyclerView.Adapter<MascotaAdapter.MascotaViewHolder>() {

    interface OnMascotaClickListener {
        fun onVerDetalles(mascota: Mascota)
    }

    fun updateList(newList: MutableList<Mascota>) {
        this.mascotaList = newList
        notifyDataSetChanged()
    }

    fun setModoEdicion(modoEdicion: Boolean) {
        this.modoEdicion = modoEdicion
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MascotaViewHolder {
        val view: View = LayoutInflater.from(context)
            .inflate(R.layout.item_mascota, parent, false)
        return MascotaViewHolder(view)
    }

    override fun onBindViewHolder(holder: MascotaViewHolder, position: Int) {
        val mascota = mascotaList[position]

        holder.txtNombre.text = mascota.getNombre()
        holder.txtEdad.text = "Edad: ${mascota.getEdad()}"
        holder.txtSexo.text = "Sexo: ${mascota.getSexo()}"
        holder.txtDescripcion.text = mascota.getDescripcion()

        val imageUrl = mascota.getImageUrl()

        if (!imageUrl.isNullOrEmpty()) {
            Glide.with(context)
                .load(imageUrl)
                .placeholder(R.drawable.ic_launcher_foreground)
                .error(R.mipmap.mascota1)
                .centerCrop()
                .into(holder.imgMascota)
        } else {
            holder.imgMascota.setImageResource(R.drawable.ic_launcher_foreground)
        }

        holder.btnDetalles.text = if (modoEdicion) "Editar" else "Ver Detalles"

        holder.btnDetalles.setOnClickListener {
            listener?.onVerDetalles(mascota)
        }
    }

    override fun getItemCount(): Int {
        return mascotaList.size
    }

    class MascotaViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        val imgMascota: ImageView = itemView.findViewById(R.id.imgMascota)
        val txtNombre: TextView = itemView.findViewById(R.id.txtNombreMascota)
        val txtEdad: TextView = itemView.findViewById(R.id.txtEdadMascota)
        val txtSexo: TextView = itemView.findViewById(R.id.txtSexoMascota)
        val txtDescripcion: TextView = itemView.findViewById(R.id.txtDescripcionMascota)
        val btnDetalles: Button = itemView.findViewById(R.id.btnVerDetalles)
    }
}